packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -80,7 +88,7 @@ const useLazyLoadCategories = () => {
         enabled: onClient && ids?.length > 0
     })
     // make sure all queries is successful before returning
-    if (onClient && queries?.every((query) => query.isSuccess)) {
+    if (onClient) {
         const dataArray = queries.map((queries) => queries.data)
```

Comment:
dataArray = queries.map((queries) => queries.data).filter(Boolean) will make sure this array does not contain any undefined. Then we dont have to guard the array in the mergeArrays